src/bin/stratis_dumpmetadata.rs

```diff
@@ -0,0 +1,58 @@
+use std::{
+    env,
+    fs::OpenOptions,
+    io::{Seek, SeekFrom},
+    process,
+    vec::Vec,
+};
+
+use serde_json::Value;
+
+use libstratis::engine::BDA;
+
+fn run(devpath: &str) -> Result<(), String> {
+    let mut devfile = OpenOptions::new()
+        .read(true)
+        .open(&devpath)
+        .map_err(|the_io_error| format!("Error opening device: {}", the_io_error))?;
+
+    let bda = BDA::load(&mut devfile)
+        .map_err(|bda_load_error| format!("BDA detected but error found: {}", bda_load_error))?
+        .ok_or_else(|| "No Stratis BDA metadata found".to_string())?;
+    println!("{:#?}", bda);
+
+    devfile
+        .seek(SeekFrom::Start(0))
+        .map_err(|seek_err| format!("Error during seek: {}", seek_err))?;
+
+    let loaded_state = bda
+        .load_state(&mut devfile)
+        .map_err(|stateload_err| format!("Error during load state: {}", stateload_err))?
+        .ok_or_else(|| "No BDA state found".to_string())?;
```

Comment:
Ok! We need to branch right here, depending on whether ```loaded_state``` is ```None``` or ```Some(..)```. So, finish the computation of ```loaded_state``` so that it is an ```Option```, like:

```
let loaded_state = bda
        .load_state(&mut devfile)
        .map_err(|stateload_err| format!("Error during load state: {}", stateload_err))?;
```

now,

```
if let Some(loaded_state) = loaded_state {
  // all the computations you've already constructed for when
  // loaded_state is Some should go on this branch
} else {
  // Different output for when loaded_state is None.
  // Why is this possible? With very many devices in a pool, the
  // metadata may only be written to a selected set, Or, the pool
  // could have gone down before the metadata could be written.
  // Either way, it's a valid possibility.
}
```

Why not use the equivalent

```
match loaded_state {
  Some(loaded_state) => { // if branch stuff },
  None => { // else branch stuff },
}
```

instead of the ```if-let``` expression, as they are completely equivalent? clippy prefers ```if-let``` and the resulting indentation is slightly easier on the eyes.